daemon/list.go

```diff
@@ -499,14 +499,14 @@ func includeContainerInList(container *container.Container, ctx *listContext) it
 	networkExist := fmt.Errorf("container part of network")
 	if ctx.filters.Include("network") {
 		err := ctx.filters.WalkValues("network", func(value string) error {
-			if _, ok := container.NetworkSettings.Networks[value]; ok {
-				return networkExist
-			}
 			for _, nw := range container.NetworkSettings.Networks {
 				if nw.EndpointSettings == nil {
 					continue
 				}
-				if nw.NetworkID == value {
+				if strings.HasPrefix(nw.NetworkID, value) {
+					return networkExist
+				}
+				if strings.HasPrefix(container.HostConfig.NetworkMode.NetworkName(), value) {
```

Comment:
Yes, I think we should not do a _partial_ match on names (at least not for now). I compared with filtering by `volume`, and that also does a _full_ match only;

```bash

$ docker volume create vol1
vol1
$docker volume create vol11
vol11
$ docker run -dit -v vol1:/foo busybox
8ce3ed958ecd9b6e2eeb63246f5b2626838a451bc57c711bc2e4c5b8530b6f38
$ docker run -dit -v vol11:/foo busybox
3ad4fe7692514e4cccfdba36c763c936b1b7729bbc46d543dda989076b36454b

$ docker ps --filter volume=vol1
CONTAINER ID        IMAGE               COMMAND             CREATED             STATUS              PORTS               NAMES
8ce3ed958ecd        busybox             "sh"                15 seconds ago      Up 14 seconds                           hardcore_mccarthy

$ docker ps --filter volume=vol11
CONTAINER ID        IMAGE               COMMAND             CREATED              STATUS              PORTS               NAMES
3ad4fe769251        busybox             "sh"                About a minute ago   Up About a minute                       optimistic_jones
```

Note that doing partial matching on _ID_ is ok, because that's what we do in other places